### DISCUSSION TITLE
Train each small crops of the wide-area image

### DISCUSSION THREAD

**Discussion by Mstfakts** [facebookresearch/detectron2#2943] [Q&A]
Hello,

I have read many articles about detecting objects from wide-area images. Most of them train the model by cropping the image. Some of them equally crops, some of them crops regarding some rule. However, I do not want to modify my training dataset. I want to write a function that crops the images by following some rules, but this will be done on the fly. I mean, I will not create or modify my dataset, I want to crop images during training and use all crops during training. 

Let's say that an image can be divided into 4 parts. I want to feed the model with 1 wide image, and the model should create 4 crops from the image and use all crops during training. I will first take the first part, then the second part, the third part, and the fourth part. After these crops, it will take the next image. 

How can I do that? Should I change the data loader or write a new data augmentation?

This is what I tried;
- I wrote a data augmentation that crops image but it literally only crops. I could not modify it to be able to train with all crops.

**Comment by ninafiona** [comment]
I want to do the same thing, cropping the image using SAHI (https://github.com/obss/sahi).

Have you been able to implement it without adjusting the training set?
